app/vmselect/promql/transform.go

```diff
@@ -420,7 +420,7 @@ func transformBucketsLimit(tfa *transformFuncArg) ([]*timeseries, error) {
 		mn.CopyFrom(&ts.MetricName)
 		mn.RemoveTag("le")
 		b = marshalMetricNameSorted(b[:0], &mn)
-		k := bytesutil.InternBytes(b)
+		k := bytesutil.ToUnsafeString(b)
```

Comment:
This transformation is unsafe, since `k` may be used as a new key in the map `m`. In this case the saved key shouldn't change over time, while in our case the `k` changes with every new loop because the underlying `b` is changed at the line 422. I fixed this in the commit 97a7128e4b4d2f04aa68cfbd93c785f8e88cdd4f .